src/pkg/packager/deploy.go

```diff
@@ -118,39 +148,86 @@ func (p *Packager) deployComponents() (deployedComponents []types.DeployedCompon
 		return deployedComponents, fmt.Errorf("unable to generate the value template: %w", err)
 	}
 
+	// Check if this package has been deployed before and grab relevant information about already deployed components
+	if p.generation == 0 {
+		p.generation = 1 // If this is the first deployment, set the generation to 1
+	}
+
+	// Fetch any installed Helm charts for this package
+	var installedCharts []types.InstalledChart
+	if p.cluster != nil {
+		installedCharts, err = p.cluster.GetInstalledCharts(p.cfg.Pkg.Metadata.Name)
+		if err != nil {
+			message.Debugf("Unable to fetch installed Helm charts for package '%s': %s", p.cfg.Pkg.Metadata.Name, err.Error())
+		}
+	}
+
+	// Process all the components we are deploying
 	for _, component := range componentsToDeploy {
-		var charts []types.InstalledChart
 
+		// Update the component status to "Deploying" and increment ObservedGeneration
+		deployedComponent := types.DeployedComponent{Name: component.Name, Status: types.ComponentStatusDeploying, ObservedGeneration: p.generation}
+		deployedComponents = append(deployedComponents, deployedComponent)
+
+		idx := len(deployedComponents) - 1
+
+		// Account for any previously installed Helm Charts before updating the package secret.
+		// This prevents us from potentially overwriting any existing installed charts data being tracked in the package secret.
+		if installedCharts != nil {
+			deployedComponents[idx].InstalledCharts = installedCharts
```

Comment:
This is just going to add all helm charts for a _package_ to each component...  Pretty sure that's not the logic we want.

Reply:
Changed `GetInstalledCharts` to `GetInstalledChartsForComponent` so that only installed charts for the specific component we are deploying will be added.